torch_xla/csrc/xla_graph_executor.cpp

```diff
@@ -1329,8 +1329,10 @@ XLAGraphExecutor::CompilationResult XLAGraphExecutor::Compile(
         torch::lazy::Output(ir_value.node.get(), ir_value.index));
     lowering_ctx.AddResult(root);
   }
+  // Always execute sharded when running in SPMD mode
+  bool is_sharded = (coll.device == GetVirtualDevice());
   // Annotate HLO sharding selectively in the compuation.
-  bool is_sharded = ShardingUtil::SetHloSharding(&lowering_ctx);
+  ShardingUtil::SetHloSharding(&lowering_ctx);
```

Comment:
can we add a test for non-sharding execution with spmd mode?

Reply:
It looks like we already have some coverage here: https://github.com/pytorch/xla/blob/4f378e694a68192fd634a070a3485c73afab1c71/test/spmd/test_xla_virtual_device.py#L55

I'll sync up with @alanwaketan on the issue he encountered and use that as a test case

Reply:
Added a test case which induces the error without this patch

Reply:
Sorry for missing your messages, but the `Unknown device: SPMD:0` is the right error.